OpenUpSA/wazimap-ng-ui
# powered by wazimap

## Description
* narrow enough to fit inside rich data in page nav
* hidden when rich data in page nav is minimised so that it doesn't overlap with rich data content
* data mapper and point mapper menus have space at the bottom so that it doesn't overlap their content, their scroll space ends above it

## Related Issue
https://wazimap.atlassian.net/browse/WNCM-555

## How is it tested automatically?
added `__tests__/gui/watermark.feature`

## How should a reviewer test it locally
* confirm that the watermarks are always in the same position
* confirm that in the data mapper and the point mapper, the scroll area ends above the watermark. and the watermark is always visible
* confirm that in the rich data panel, if you collapse the categories menu the watermark is not visible anymore

## Screenshots
* rich data
![image](https://user-images.githubusercontent.com/53019884/206901866-d331a56a-a6a1-466b-95ec-c111850364d9.png)

* rich data with the categories menu collapsed
![image](https://user-images.githubusercontent.com/53019884/206901896-79be4081-838f-459e-b221-a7f3af3d099e.png)

* data mapper
![image](https://user-images.githubusercontent.com/53019884/206901918-e893cfda-5a93-46a3-b8b7-ad712d24e40f.png)

* data mapper with the scrollbar
![image](https://user-images.githubusercontent.com/53019884/206901969-6a9cd476-9df4-4267-96e5-b1ee6eca7ab4.png)

* point mapper
![image](https://user-images.githubusercontent.com/53019884/206902003-0171d884-811a-4f20-abfc-93b72217527e.png)

* point mapper with the scrollbar
![image](https://user-images.githubusercontent.com/53019884/206902027-08f5472e-72b7-48bf-928d-611a4d2b59fd.png)


## Changelog

### Added
* `src/js/ui_components/watermark.js` watermark react component
* `src/custom-css/general.scss` for site-wide css changes

### Updated
* `src/js/profile/profile_loader.js` to add the watermark to rich data
* `src/js/map/maps.js` to add the watermark to the map
* `src/js/elements/point_tray/tray.js` to add the watermark to the point mapper
* `src/js/elements/data_mapper/menu.js` to add the watermark to the data mapper
* `src/js/webflow/import.js` to set up the watermark elements in the index.html


### Removed


## Checklist


### Pull Request


### Code Quality


### Testing
